### PR TITLE
add netcoreapp3.0 global tool

### PR DESCRIPTION
**Pull request by ctaggart**
Based on #279 from @onovotny, but uses latest preview 3 and sets the global.json just on the Azure Pipelines build.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -35,12 +35,16 @@ jobs:
     displayName: Set build number
 
   - task: DotNetCoreInstaller@0
-    displayName: Install .NET Core SDK 2.1.500
+    displayName: Install .NET Core SDK
     inputs:
       packageType: sdk
-      version: 2.1.500
+      version: 3.0.100-preview3
+  - script: dotnet new globaljson --sdk-version 3.0.100-preview3
```
Comment (onovotny) [inline azure-pipelines.yml]: Neat. you can also use `3.0.100-preview` as the version as the lookup rules will find the highest matching preview. That means you won't forget to update it later.

**Comment by ctaggart** [comment]
I'm not sure what this error is from:

image

I know that switching from TargetFramework to TargetFrameworks adds an additional folder. May be something in the gulpfile needs updating.

File: src/nbgv/nbgv.csproj [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <ToolCommandName>nbgv</ToolCommandName>
     <PackAsTool>True</PackAsTool>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFrameworks>netcoreapp2.1;netcoreapp3.0</TargetFrameworks>
```
Comment (ctaggart) [inline src/nbgv/nbgv.csproj]: If you are missing one of the target frameworks, does it still load? Hoping so.

File: src/Directory.Build.props [hunk 3]
```diff
@@ -1,8 +1,6 @@
 <Project>
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <BaseIntermediateOutputPath>$(MSBuildThisFileDirectory)..\obj\$(MSBuildProjectName)\</BaseIntermediateOutputPath>
-    <OutputPath>$(MSBuildThisFileDirectory)..\bin\$(MSBuildProjectName)\$(Configuration)\</OutputPath>
```
Comment (ctaggart) [inline src/Directory.Build.props]: This combined with TargetFrameworks does not work. It is easier to just `dotnet publish`.

File: src/nbgv/nbgv.csproj [hunk 4]
```diff
@@ -4,14 +4,12 @@
...
     <RootNamespace>Nerdbank.GitVersioning.Tool</RootNamespace>
     <Description>A .NET Core Tool that can install, read and set version information based on git history, using Nerdbank.GitVersioning.</Description>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="NuGet.PackageManagement" Version="4.9.3" NoWarn="NU1701" />
-    <PackageReference Include="NuGet.Resolver" Version="4.9.3" />
```
Comment (ctaggart) [inline src/nbgv/nbgv.csproj]: These were only used by `GetLatestPackageVersionAsync`. They are netframework only and cause a bunch of dependency problems. Hopefully there is an easier way with NuGet 5.0.

File: azure-pipelines.yml (hunk 1 above)
Comment (ctaggart) [reply azure-pipelines.yml]: This doesn't work because of the `netcoreapp3.0` dependency, so I removed it.

**Pull request closed without merging**